src/perfoutputwidgetkonsole.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+  perfoutputwidgetkonsole.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "perfoutputwidgetkonsole.h"
+
+#include <QEvent>
+#include <QKeyEvent>
+#include <QStandardPaths>
+#include <QTemporaryFile>
+#include <QVBoxLayout>
+
+#include <KParts/ReadOnlyPart>
+#include <KParts/kde_terminal_interface.h>
+#include <KService>
+
+QString PerfOutputWidgetKonsole::m_tailPath;
+
+PerfOutputWidgetKonsole::PerfOutputWidgetKonsole(KParts::ReadOnlyPart* part, QWidget* parent)
+    : PerfOutputWidget(parent)
+    , m_konsolePart(part)
+{
+    auto layout = new QVBoxLayout(this);
+    setLayout(layout);
+
+    addPartToLayout();
+}
+
+PerfOutputWidgetKonsole::~PerfOutputWidgetKonsole() = default;
+
+PerfOutputWidgetKonsole* PerfOutputWidgetKonsole::create(QWidget* parent)
+{
+    m_tailPath = findTail();
+    if (m_tailPath.isEmpty()) {
+        return nullptr;
+    }
+
+    auto part = createPart();
+    if (!part) {
+        return nullptr;
+    }
+
+    const auto terminalInterface = qobject_cast<TerminalInterface*>(part);
```

Comment:
why not pass that TerminalInterface directly to the PerfOutputWidgetKonsole? that would remove the need for casting below at...

Reply:
I still need the KParts::ReadOnlyPart to add the widget to the layout and I can't cast the TerminalInterface back to a KParts::ReadOnlyParts since it is missing the Q_OBJECT macro.
I guess adding a new member variable pointing to the same object would be the only option to remove the need for casting.

Reply:
understood, I fogot the fact that this is using multiple inheritance with Q_INTERFACES